packages/vscode-extension/src/envTree.ts

```diff
@@ -44,9 +52,45 @@ export async function registerEnvTreeHandler(): Promise<Result<Void, FxError>> {
           isCustom: false,
           description:
             item === activeEnv ? StringResources.vsc.commandsTreeViewProvider.acitve : "",
+          expanded: true,
         },
       ]);
-    });
+    }
+
+    for (const item of envNamesResult.value) {
+      await updateCollaboratorList(item);
+    }
   }
   return ok(Void);
 }
+
+export async function updateCollaboratorList(env: string): Promise<void> {
+  if (environmentTreeProvider && isRemoteCollaborateEnabled()) {
+    let userList: TreeItem[] = [];
+
+    const parentCommand = environmentTreeProvider.findCommand("fx-extension.environment." + env);
+    if (parentCommand) {
+      const loginStatus = await AppStudioLogin.getInstance().getStatus();
+      if (loginStatus.status == signedIn) {
+        const canAddCollaborator = await checkPermission(env);
+        parentCommand.contextValue = canAddCollaborator
+          ? "environmentWithPermission"
+          : "environment";
+        if (isRemoteCollaborateEnabled()) {
+          userList = await listCollaborator(env);
+        }
+      } else {
+        userList = [
+          {
+            commandId: `fx-extension.listcollaborator.${env}`,
+            label: "Login M365 account to view all collaborators",
```

Comment:
Better move to strings.json if the string is available to users for later localization.

Reply:
OK, thanks for your tip, will update

Reply:
Updated